src/apps/backend/modules/application/worker/internal/worker_manager.py

```diff
@@ -0,0 +1,141 @@
+import uuid
+from typing import Optional
+
+from temporalio.client import Client, WorkflowExecutionStatus, WorkflowHandle
+from temporalio.service import RetryConfig
+
+from modules.application.worker.errors import (
+    WorkerAlreadyCancelledError,
+    WorkerAlreadyCompletedError,
+    WorkerAlreadyTerminatedError,
+    WorkerClassNotRegisteredError,
+    WorkerClientConnectionError,
+)
+from modules.application.worker.types import RunWorkerCronParams, RunWorkerParams, SearchWorkerByIdParams
+from modules.config.config_service import ConfigService
+from workers.worker_registry import WORKER_MAP
+
+
+class WorkerManager:
+    CLIENT: Optional[Client] = None
+
+    @staticmethod
+    async def _get_client() -> Client:
+        if not WorkerManager.CLIENT:
+            try:
+                WorkerManager.CLIENT = await Client.connect(
+                    ConfigService.get_string("TEMPORAL_SERVER_ADDRESS"), retry_config=RetryConfig(max_retries=3)
+                )
+
+            except RuntimeError:
+                raise WorkerClientConnectionError(server_address=ConfigService.get_string("TEMPORAL_SERVER_ADDRESS"))
+
+        return WorkerManager.CLIENT
+
+    @staticmethod
+    async def _get_worker_status(handle: WorkflowHandle) -> Optional[WorkflowExecutionStatus]:
+        info = await handle.describe()
+        return info.status
+
+    @staticmethod
+    async def get_worker_details(params: SearchWorkerByIdParams) -> dict:
+        client = await WorkerManager._get_client()
+
+        runs = []
+
+        async for info in client.list_workflows(f"WorkflowId = '{params.id}'", limit=params.runs_limit):
+            handle = client.get_workflow_handle(workflow_id=params.id, run_id=info.run_id)
+            info = await handle.describe()
+
+            result = None
+            if info.status and info.status == WorkflowExecutionStatus.COMPLETED:
+                history = await handle.fetch_history()
+                result_event = history.events[-1]
+                result_data = result_event.workflow_execution_completed_event_attributes.result.payloads[0].data
+                result = result_data.decode("utf-8")
+
+            runs.append(
+                {
+                    "run_id": info.run_id,
+                    "status": info.status,
+                    "result": result,
+                    "start_time": info.start_time,
+                    "close_time": info.close_time,
+                }
+            )
+
+        handle = client.get_workflow_handle(params.id)
+        info = await handle.describe()
+        return {
+            "worker_id": info.id,
+            "status": info.status,
+            "runs": runs,
+            "start_time": info.start_time,
+            "close_time": info.close_time,
+            "task_queue": info.task_queue,
+            "worker_type": info.workflow_type,
+        }
```

Comment:
I am not able to understand this function.

Is this trying to return all completed instances of a given workflow? If yes, the function name is quite misleading. Furthermore I think we can break into smaller functions to describe "what" it does instead of "how" it does